Pop up window

tomasesquivelgc opened the pull request:
## What's new in this branch:

### Pop-up window
 - Every project's button now creates a div with JS.
 - This div generates h3, ul, image and p elements inside of it.
 - The text in each element varies from project to project
 - This pop-up window is responsive to size.

tomasesquivelgc committed:
pop-up-window create

js object project add
yoga project add (as test)
js function to create div add, append to button

tomasesquivelgc committed:
change z-index

tomasesquivelgc committed:
background for newsection create

newsection now has width of 100% with blur
inside newsection is a div with 90% width and white background
image rescaled
pop-up image css add

tomasesquivelgc committed:
newsection style

pop-up-window css add
object art-printing create

tomasesquivelgc committed:
objects add

all corresponding objects add
ul li p elements give new class
newsection ul li p elements also new class add
new section buttons image add

tomasesquivelgc committed:
syntax fix

tomasesquivelgc committed:
functions mention to pass linters

Zuhaib042 reviewed (commented):
Hi @tomasesquivelgc 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair-programming activity and I dont see any commit from your partner. kindly consider adding at least one commit from your partner.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

lerfast committed:
Update JS and Readme files

Making changes to some JS variables.
Adding me as a contributor

Shedrack-Sunday commented on script.js:
```diff
@@ -4,10 +4,76 @@ openButton.onclick = () => {
     y.style.display = 'none';
   };
 });
+
+const projects = {
+  yoga: {
+    title: 'Multi-Post Stories',
+    technology1: 'CSS',
+    technology2: 'HTML',
+    technology3: 'Bootstrap',
+    technology4: 'Ruby',
+    image: './Normal_Button/yoga-img.jpg',
+    paragraph: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent",
+  },
```
- [x] Nice work you did here on the array to dynamically render. But here are suggested fixes:
- Your `technologies` should also be wrapped in a nested array, to avoid the repetition of keys 👍 
- Kindly include app link in the array

## Example implementation
```
const projects = [{
    name: "Sample app",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```

tomasesquivelgc replied:
Hello, thank you for your review. I am working on the arrays just like you said, but I don't have a link to anything as these are not real apps. Should I just add a dead link?

Shedrack-Sunday commented on script.js:
```diff
@@ -4,10 +4,76 @@ openButton.onclick = () => {
+  const newSection = document.createElement('div');
+  newSection.id = 'pop-up-window';
+  newSection.innerHTML = `<div class='pop-up-container'><h3 style='margin: 0;'>${projects[projectId].title}<button id='closePopUp' class='close-pop-up' onclick='closeNewSection()'><img src='./Normal_Button/Close_button.png' alt='Close menu'></button></h3>
+                          <ul><li><p class='new-section-technology'>${projects[projectId].technology1}</p></li><li><p class='new-section-technology'>${projects[projectId].technology2}</p></li><li><p class='new-section-technology'>${projects[projectId].technology3}</p></li><li><p class='new-section-technology'>${projects[projectId].technology4}</p></li></ul>
```
- You did well in this section for the technologies 👏 . However, You'll notice there is a padding missing in the box of yours, in reference to that on the figma version. You'll agree with me that the figma version has a more better user appealing view, and looks more professional 👍 . Kindly make the change based on the image below 

## Your view:
![image](https://user-images.githubusercontent.com/50990925/231592816-b77ecfca-ef44-4a5b-890e-5b2b55ee8b2f.png)

## Figma:
![image](https://user-images.githubusercontent.com/50990925/231592936-0154f62b-d8dc-4697-a9dc-b0fac052120e.png)

Shedrack-Sunday commented on script.js:
```diff
@@ -4,10 +4,76 @@ openButton.onclick = () => {
+  const container = document.getElementById('yoga-project');
+  const newSection = document.createElement('div');
+  newSection.id = 'pop-up-window';
+  newSection.innerHTML = `<div class='pop-up-container'><h3 style='margin: 0;'>${projects[projectId].title}<button id='closePopUp' class='close-pop-up' onclick='closeNewSection()'><img src='./Normal_Button/Close_button.png' alt='Close menu'></button></h3>
```
- [x] You did put in a lot of work On the pop-up on desktop, however, there happens to be a couple of issues which i highlighted below:

- As i stated on line `63`, kindly adjust the paddings on the technologies stated 
- From the figma design, there's a space between the image and paragrah. Kindly asjust that also 
- The spacing between the paragraph and the buttons is missing. Fixing this gives a more appealing view, and a better user experience 👍 


## Your view
![image](https://user-images.githubusercontent.com/50990925/231594421-3c4b6b4b-901b-4fed-862d-37147b0b4bbf.png)

## Expected from figma:
![image](https://user-images.githubusercontent.com/50990925/231594757-a10b69c0-7894-45d7-8b9d-2290e88893cb.png)

Shedrack-Sunday reviewed (changes requested):
Hi @tomasesquivelgc     👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 
![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)

## Highlights
- Proper use webpack build ✔️ 
- Nice README structure 👍 
- Moderate design 💯 
  
# Required Changes ♻️

- There's a bit content structure that defers i reference to the figma design given

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

tomasesquivelgc committed:
technologies now into one array

tomasesquivelgc committed:
render function fix

function now takes values from array inside object.
image and paragraph+buttons padding add
paragraph+button justify content space between

tomasesquivelgc committed:
syntax fix

Reem-lab commented on script.js:
```diff
@@ -4,10 +4,71 @@ openButton.onclick = () => {
+                          <ul>${liItems}</ul>
+                          <img src='${projects[projectId].image}' alt='${projects[projectId].title}' class='pop-up-image'/>
+                          <div class='text-and-buttons'><p>${projects[projectId].paragraph}</p>
+                          <div class=button-div><button class=see-project1>See Live<img src='./Normal_Button/seeLive-button.svg' alt='See live'></button><button class='see-project1'>See Source<img src='./Normal_Button/seeSource-button.svg' alt='See source'</button></div></div></div>';`;
```
- [ ]Great job working on the requested changes but  I highly recommend to add it some margin here as well since in the mobile version I think the buttons is very close to the text above there is no space to have better UI/UX kindly update this 💪🏻🤩💯✨

- your 
![image](https://user-images.githubusercontent.com/58553711/231621456-4a62ee20-e0fc-4a84-b4db-8c0cadd2e0a4.png)

- Figma
![image](https://user-images.githubusercontent.com/58553711/231621549-2a451325-816e-4bf0-8ede-abbf748eb4ff.png)

tomasesquivelgc replied:
I will do that right away. Thank you very much!

Reem-lab reviewed (approved):
# Hi Team,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯
<img src="https://media2.giphy.com/media/3o6fJ1BM7R2EBRDnxK/giphy.gif"/>

## Highlights 🔥🔥🔥🔥
- Followed GitHub flow ✅
- Great working on the functionality of popup🔥🔥
- All requested changes are done✅🤩




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tomasesquivelgc committed:
add margin to button div in mobile version only
